lighthouse-core/lib/tracehouse/trace-processor.js

```diff
@@ -62,10 +62,110 @@ class TraceProcessor {
   }
 
   /**
+   * This method sorts a group of trace events that have the same timestamp. We want to...
+   *
+   * 1. Put E events first, we finish off our existing events before we start new ones.
+   * 2. Order B/X events by their duration, we want parents to start before child events.
+   * 3. If we don't have any of this to go on, just use the position in the original array (stable sort).
+   *
```

Comment:
is it worth calling out for future readers that this is expected to usually get `tsGroupIndices` lengths on the order of 10 (and often closer to 1?). Obviously future readers can use profilers, too, but it does jump out as something pretty heavyweight to run over a whole trace when actually it's run over a pretty small part of a trace

Reply:
yeah good call, it's a good reminder we'll all need for the next poor sap to touch trace events 👍 😆